src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

```diff
@@ -181,8 +181,12 @@ void AddEntryCore(UnusedValuePreference preference, bool isUnusedLocalAssignment
         public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
 
         protected sealed override void InitializeWorker(AnalysisContext context)
-            => context.RegisterCompilationStartAction(
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
```

Comment:
Does this mean all generated code will get analyzed for unused parameters? Will we reporting diagnostics on generated files?

Reply:
No, this flag just enables analysis callbacks for generated code, but reported analyzer diagnostics in generated code are still filtered out. One needs to turn on [GeneratedCodeAnalysisFlags.ReportDiagnostics](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs,259) bit to also report diagnostics in generated code. The second unit test added in the PR explicitly ensures we don't report unused parameters if the containing symbol is generated code.